src/signatures/secp256k1.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::convert::TryInto;
+use k256::{
+    elliptic_curve::{
+        ecdh,
+        generic_array::{typenum::Unsigned, GenericArray},
+        sec1::{Coordinates, EncodedPoint, ToEncodedPoint, UntaggedPointSize},
+    },
+    Secp256k1,
+};
+use signature::{Error as SigError, Signer, Verifier};
+
+/// A byte representation of an Secp256k1 field element.
+pub type FieldBytes = k256::FieldBytes;
+
+/// Secp256k1 Shared Secret - the result of a Diffie-Hellman key exchange.
+///
+/// Each party computes this from an (ephemeral) [SecretKey] and their counterparty's [PublicKey].
+pub type SharedSecret = k256::ecdh::SharedSecret;
+
+/// Secp256k1 Public Key
+pub struct PublicKey(k256::PublicKey);
+
+// algorithm identifier used for errors
+const NAME: &str = "Secp256k1";
+
+impl PublicKey {
+    /// Load a [PublicKey] from a slice of bytes.
+    pub fn from_bytes(bytes: &[u8]) -> crate::Result<Self> {
+        assert_buffer_eq!(bytes.len(), UntaggedPointSize::<Secp256k1>::USIZE, "bytes");
+
+        EncodedPoint::from_untagged_bytes(bytes.into())
+            .decode()
+            .map(Self)
+            .map_err(|_| crate::Error::ConvertError {
+                from: "bytes",
+                to: "Secp256k1 Public Key",
+            })
+    }
+
+    /// Load a [PublicKey] from big-endian serialized coordinates.
+    pub fn from_coord(x: &[u8], y: &[u8]) -> crate::Result<Self> {
```

Comment:
Do we need "coord" representation? Why not just keep those concatenated in one field / byte array?

Reply:
I'm not sure if we absolutely need it. I can't find a usage of the `from_coord` function, but we use the `to_coord` function for conversion into a `JWK`, here: https://github.com/iotaledger/identity.rs/blob/4287c9834de8dc42fd4ff0c0e8b3a2bc822d0ec4/libjose/src/jwk/key.rs#L487
For symmetry, I suppose it makes sense to keep both from and to functions, but if there's an alternative representation that achieves the same, I think we can also use that. Note that I'm not an expert on crypto math, so feel free to tell me what you think is best.

Reply:
Right, I see. So, it's a standard way to represent secp256k1 points used by other implementations. In that case we should definitely keep it.